SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt

```diff
@@ -32,20 +34,32 @@ class ApplicationPresenter: ApplicationContract.Presenter() {
     }
 
     override fun onButtonTapped() {
+        if (chosenDepartureStation == chosenArrivalStation) return
         val dateTimeFormat = DateFormat("yyyy-MM-ddTHH:mm:ss.000")
         val timeNow: String = DateTimeTz.nowLocal().format(dateTimeFormat)
 
         val apiCall = "https://mobile-api-dev.lner.co.uk/v1/fares?originStation=$chosenDepartureStation&destinationStation=$chosenArrivalStation&noChanges=false&numberOfAdults=1&numberOfChildren=0&journeyType=single&outboundDateTime=$timeNow&outboundIsArriveBy=false"
+
         val client = HttpClient() {
             install(JsonFeature) {
                 serializer = KotlinxSerializer(Json.nonstrict)
             }
         }
 
         launch {
-            val jsonString = client.get<DepartureDetails>(apiCall)
+            lateinit var jsonString: DepartureDetails
+            try{
+                jsonString = client.get(apiCall)
+            } catch (e: Exception){
+                view!!.setLabel("API Call Failed")
+            }
             val departures: MutableList<DepartureInformation> = mutableListOf()
-            for (i in 0..4){
+            var numOfResults = 5
```

Comment:
Is the purpose of all this just to do the iteration over the results?  It seems like `jsonString.outboundJourneys` is a list object so you can iterate over that directly